anospp_analysis/plasm.py

```diff
@@ -0,0 +1,671 @@
+    haps_merged_df = process_haplotypes(hap_df, comb_stats_df, args.filter_p1, args.filter_p2)
+
+    logging.info('Checking for the presence of PLASM_TARGETS')
+    if len(haps_merged_df['target'].unique()) < 1:
+        logging.warning('Could not find both PLASM_TARGETS in hap_df')
+        sys.exit(1)
+
+    else:
+        logging.info('## run blast')
+        df_list = []
+        hap_output = []
+        for target in haps_merged_df['target'].unique():          
+            haplotype_df, new_cols = haplotype_summary(hap_df, target, args.workdir)
+            blast_df = run_blast(haps_merged_df, target, args.workdir, args.blastdb, args.filter_F1, args.filter_F2)
+            hap_div_df = haplotype_diversity(haplotype_df, target, new_cols, hap_df, blast_df, args.workdir)
+            generate_haplotype_tree(target, hap_div_df, args.workdir, args.interactive_plotting)
+            df_list.append(create_per_read_summary(blast_df, target, args.workdir))
+            hap_output.append(blast_df)
+
+        logging.info('## create a dataframe used the merged summary outputs')
+        merged_df = pd.concat(df_list, axis=1)
+        merge_and_export(samples_df, merged_df, args.workdir)
+        df_all = process_results(args.filter_p1, args.filter_p2, args.workdir, args.outdir)
+        merged_hap_df = pd.concat(hap_output, axis=0)[['sample_id', 'hap_id', 'target', 'consensus', 'pident']].copy().set_index('sample_id')
+        merged_hap_df.to_csv(f'{args.outdir}/Plasmodium_haplotypes_for.tsv', sep='\t')
+
+        logging.info('merging the samples(meta) dataframe with the stats dataframe and creating plots')
+        meta_df_all = pd.merge(samples_df.set_index('sample_id'), df_all, left_index =True, right_index=True, how='left')
+        generate_plots(meta_df_all, haps_merged_df, args.workdir, args.reference, args.interactive_plotting)
+
+        logging.info('## completed the plasm program!!!')
+
+
+def main():
+    
+    parser = argparse.ArgumentParser("QC for ANOSPP sequencing data")
+    parser.add_argument('-a', '--haplotypes', help='Haplotypes tsv file', required=True)
+    parser.add_argument('-m', '--manifest', help='Samples manifest tsv file', required=True)
+    parser.add_argument('-s', '--stats', help='DADA2 stats tsv file', required=True)
+    parser.add_argument('-b', '--blastdb', help='Blast db prefix. Default: ref_v1.0/plasmomito_P1P2_DB_v1.0', 
+        default='ref_v1.0/plasmomito_P1P2_DB_v1.0', required=True)
+    parser.add_argument('-o', '--outdir', help='Output directory. Default: qc', default='plasm')
+    parser.add_argument('-w', '--workdir', help='Working directory. Default: work', default='work')
+    parser.add_argument('-d', '--reference', help='Blast db prefix. Default: ref_v1.0', default='ref_v1.0')
+    parser.add_argument('-c', '--readcutoff', help='Read cutoffs. Default: 10', default=10)
+    parser.add_argument('-q', '--filter_p1', help='Plasmodium genus haplotype filter for P1. Default: 10', default=10)
```

Comment:
is it cutoff for confident infection? or removing the haplotype altogether?

Reply:
one other possibility is to specify all four cutoffs in one parameter as `--filter_cutoffs 10,10,10,10` - then parse out values

Reply:
help message smth like `read count cutoffs for confident infection: P1 whole genus, P2 whole genus, P1 falciparum, P2 falciparum. Default: 10,10,10,10`

Parse with `args.filter_cutoffs.split(',')`

Reply:
actually, might be best to specify as two params:
- `--hard_filters  - remove all sequences supported by less than X reads for P1 and P2. Default: 10,10`
- `--soft_filters - mark as non-confident all sequences of predominant haplotype supported by less than X reads for P1 and P2. Default: 50,100`